mercure.rst

```diff
@@ -0,0 +1,564 @@
+However, serializing the resource in a hypermedia formats such as JSON-LD,
+Atom, HTML or XML is recommended.
+
+Subscribing
+~~~~~~~~~~~
+
+Subscribing to updates in JavaScript is straightforward:
+
+.. code-block:: html
+
+    <script>
+    const es = new EventSource('http://localhost:3000/hub?topic=' + encodeURIComponent('http://example.com/books/1'));
+    es.onmessage = e => {
+        // Will be called every time an update is published by the server
+        console.log(JSON.parse(e.data));
+    }
+    </script>
+
+Mercure also allows to subscribe to several topics,
+and to use URI Templates as patterns:
+
+.. code-block:: html
+
+    <script>
+    const u = new URL('http://localhost:3000/hub'); // URL is a built-in JavaScript class to manipulate URLs
+    u.searchParams.append('topic', 'http://example.com/books/1');
+    u.searchParams.append('topic', 'http://example.com/books/2'); // Subscribe to updates of several Book resources
+    u.searchParams.append('topic', 'http://example.com/reviews/{id}'); // All Review resources will match this pattern
+
+    const es = new EventSource(u);
+    es.onmessage = e => {
+        console.log(JSON.parse(e.data));
+    }
+    </script>
+
+.. tip::
+
+    Google Chrome DevTools natively integrate a `practical UI`_ displaying in live
+    the received events:
+
+    .. image:: /_images/mercure/chrome.png
+
+    To use it:
+    * open the DevTools
+    * select the "Network" tab
+    * click on the request to the Mercure hub
+    * click on the "EventStream" sub-tab.
+
+Async dispatching
+-----------------
+
+Instead of calling the `Publisher` service directly, you can also let Symfony
+dispatching the updates asynchronously thanks to the provided integration with
+the Messenger component.
+
+First, be sure :doc:`to install the Messenger component </messenger>`
+and to configure properly a transport (if you don't, the handler will
+be called synchronously).
+
+Then, dispatch the Mercure `Update` to the Messenger's Message Bus,
+it will be handled automatically:
+
+.. code-block:: php
+
+    // src/Controller/PublishController.php
+    namespace App\Controller;
+
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Messenger\MessageBusInterface;
+    use Symfony\Component\Mercure\Update;
+
+    class PublishController
+    {
+        public function __invoke(MessageBusInterface $bus): Response
+        {
+            $update = new Update(
+                'http://example.com/books/1',
+                json_encode(['status' => 'OutOfStock'])
+            );
+
+            // Sync, or async (RabbitMQ, Kafka...)
+            $bus->dispatch($update);
+
+            return new Response('published!');
+        }
+    }
+
+Discovery
+---------
+
+The Mercure protocol comes with a discovery mechanism.
+To leverage it, the Symfony application must expose the URL of the Mercure Hub
+in a `Link` HTTP header.
+
+.. image:: /_images/mercure/discovery.png
+
+*Courtesy of Mercure.rocks*
```

Comment:
Not sure about this "courtesy" message. We never do this ... and we already mention "Mercure.rocks" several times in this article.

Reply:
@javiereguiluz I'm ok to remove these mentions. This image is AGPL (as the code of the hub), not sure if it's an issue or not.